### PR TITLE
Allow to use `concat` function with empty arguments 

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow to use `concat` function with empty arguments 
``` sql 
:) select concat(); 

SELECT concat()

Query id: 675fdc33-e454-4ed1-bec3-686f2c7beea4

   ┌─concat()─┐
1. │          │
   └──────────┘

1 row in set. Elapsed: 0.001 sec. 
```

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit bb8fc59d7c3b0dcfb59d1e31bafb11bd302211d7 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65887/bb8fc59d7c3b0dcfb59d1e31bafb11bd302211d7/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/65887/bb8fc59d7c3b0dcfb59d1e31bafb11bd302211d7/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: docs/en/sql-reference/functions/string-functions.md [hunk 1]
```diff
@@ -571,15 +571,13 @@ SELECT upper('clickhouse');
 
 Converts a string to lowercase, assuming that the string contains valid UTF-8 encoded text. If this assumption is violated, no exception is thrown and the result is undefined.
 
-Does not detect the language, e.g. for Turkish the result might not be exactly correct (i/İ vs. i/I).
```
Comment (vdimir) [inline docs/en/sql-reference/functions/string-functions.md:574]: Why is it removed?

File: src/Functions/concat.cpp [hunk 2]
```diff
@@ -46,25 +46,30 @@ class ConcatImpl : public IFunction
 
     DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
     {
-        if (arguments.size() < 2)
-            throw Exception(
-                ErrorCodes::TOO_FEW_ARGUMENTS_FOR_FUNCTION,
-                "Number of arguments for function {} doesn't match: passed {}, should be at least 2",
-                getName(),
-                arguments.size());
+        if (arguments.size() == 1)
```
Comment (vdimir) [inline src/Functions/concat.cpp:49]: Can we return the same string if there's only one argument?

Comment (taiyang-li) [reply src/Functions/concat.cpp:49]: No. We can't. Refer to https://github.com/ClickHouse/ClickHouse/pull/57000

File: docs/en/sql-reference/functions/string-functions.md (hunk 1 above)
Comment (rschu1ze) [reply docs/en/sql-reference/functions/string-functions.md:574]: Leftover of https://github.com/ClickHouse/ClickHouse/pull/65761

File: docs/en/sql-reference/functions/string-functions.md [hunk 3]
```diff
@@ -736,13 +734,13 @@ concat(s1, s2, ...)
 
 **Arguments**
 
-At least one value of arbitrary type.
+Values of arbitrary type. Empty argument is also allowed.
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/string-functions.md:737]: Suggest to omit the second sentence, the first sentence implicitly assumes it (and it is a slightly silly special case without much practical relevance anyways).

File: docs/en/sql-reference/functions/string-functions.md [hunk 4]
```diff
@@ -736,13 +734,13 @@ concat(s1, s2, ...)
...
 
 Arguments which are not of types [String](../data-types/string.md) or [FixedString](../data-types/fixedstring.md) are converted to strings using their default serialization. As this decreases performance, it is not recommended to use non-String/FixedString arguments.
 
 **Returned values**
 
-The String created by concatenating the arguments.
+The String created by concatenating the arguments. If there are no arguments, the function returns empty string.
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/string-functions.md:743]: Same (we could omit the second sentence)

**Pull request merged**